Sources/SourceKitLSP/Rename.swift

```diff
@@ -186,12 +186,8 @@ fileprivate struct SyntacticRenamePiece {
     else {
       return nil
     }
-    guard
-      let start = snapshot.positionOf(zeroBasedLine: line - 1, utf8Column: column - 1),
-      let end = snapshot.positionOf(zeroBasedLine: endLine - 1, utf8Column: endColumn - 1)
-    else {
-      return nil
-    }
+    let start = snapshot.positionOf(zeroBasedLine: line - 1, utf8Column: column - 1)
+    let end = snapshot.positionOf(zeroBasedLine: endLine - 1, utf8Column: endColumn - 1)
```

Comment:
And in this case I really don't think it makes sense to do closest match (really anywhere we're making modifications), so... I'm more leaning towards having `nil` variants 😅

Reply:
I would prefer to not have `nil` returning position conversion variants. The cases such a variant would catch are:
- If we have some offset error in the line and the token to rename happens to be right at the beginning or end of the file so that the line kicks it out of the valid line range. But then rename in the middle of the file is already very broken so I doubt that it would help much
- Same argument for column indices.

We already have so many position conversion functions and I don’t want to add even more. Actually, I think it’s more likely that we would end up calling the nil returning variant somewhere in the future where recovery would have been sensible. And, in practice I’ve never seen us hit position conversion to fail, and the code is easier to maintain if you don’t have to deal with failed conversions.